lark/parser_frontends.py

```diff
@@ -49,11 +47,11 @@ def __init__(self, lexer_conf, parser_conf, options, parser=None):
         if parser:  # From cache
             self.parser = parser
         else:
-            create_parser = {
-                'lalr': create_lalr_parser,
-                'earley': create_earley_parser,
-                'cyk': CYK_FrontEnd,
-            }[parser_conf.parser_type]
+            create_parser = _parser_creators.get(parser_conf.parser_type)
```

Comment:
This previously would always find a value for the given key, but then would raise an exception when `NotImplemented` was used as a `Callable` (which isn't supported). Now there is a descriptive exception when the given key isn't supported.